core/shared/src/main/scala/spire/math/poly/Monomial.scala

```diff
@@ -0,0 +1,375 @@
+package spire.math.poly
+
+import scala.collection.SortedMap
+import scala.annotation.tailrec
+import scala.reflect._
+import scala.{specialized => spec}
+import spire.algebra._
+import spire.implicits._
+import spire.math._
+import scala.util.matching.Regex
+
+
+/**
+ *  A monomial is the product of a coefficient and a list of
+ *  variables (Char as symbol) each to an integer power.
+*/
+class Monomial[@spec(Double) C: ClassTag: Order] private[spire] (val coeff: C, val vars: SortedMap[Char, Int])
```

Comment:
I'm not sure on either question (`ClassTag` or `Order`).  They were in the original PR.  I can't imagine why Monomial would need to order anything based on coefficient.  Let me see if I can remove both of them.  If I can't, I'll explain why not.